scio-core/src/main/scala/com/spotify/scio/ScioContext.scala

```diff
@@ -357,7 +358,8 @@ class ScioContext private[scio] (val options: PipelineOptions,
       TestDataManager.closeTest(testId.get, result)
     }
 
-    if (this.isTest || this.optionsAs[ScioOptions].isBlocking) {
+    if (this.isTest || (this.optionsAs[ScioOptions].isBlocking &&
+      Option(this.optionsAs[ScioOptions].getJobTimeout).isEmpty)) {
```

Comment:
I think we should check if the option was set and if so ... we should enforce `result.waitUntilFinish(<duration>)` in case the job itself is not calling it.

Reply:
you mean something like

```
    if (Option(this.optionsAs[ScioOptions].getBlockFor).isDefined) {
      result.waitUntilFinish(duration)
    } else if (this.isTest || (this.optionsAs[ScioOptions].isBlocking)) {
      result.waitUntilDone()
    } else {
      result
    }
```
?
the only problem is we don't have the value of `cancelJob` in that scope so we don't know whether to call `result.waitUntilFinish(duration, cancelJob = true)` -- and if we added it I feel like `ScioContext#close()` would become functionally the same as `ScioResult#waitUntilFinish()`. what do you think?